### PR TITLE
[SDL-0296] GetCapabilities script

### PR DESCRIPTION
**Pull request by aderiabin**
This PR is **[ready]** for review.

### Summary
Check GetCapabilities processing 

### ATF version
develop

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/UpdateVideoStreamingCapabilities/001_3_GetCapabilities_in_range_minsize_maxsize_array.lua [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  common.Title("Preconditions")
+  common.Step("Clean environment", common.preconditions)
+  common.Step("Set HMI Capabilities", common.setHMICapabilities, { common.getVideoStreamingCapability(value) })
+  common.Step("Start SDL, HMI, connect Mobile, start Session", common.start, { common.hmiDefaultCapabilities })
+  common.Step("RAI", common.registerAppWOPTU)
+  common.Step("Activate App", common.activateApp)
+
+  common.Title("Test")
+  common.Step("GetSystemCapability in range " .. parameter .. " " .. value, common.getSystemCapability,
+    { false, appSessionId, common.getVideoStreamingCapability(value) })
```
Comment (GetmanetsIrina) [inline test_scripts/UpdateVideoStreamingCapabilities/001_3_GetCapabilities_in_range_minsize_maxsize_array.lua:25]: Please define `false` value using a local variable

File: test_scripts/UpdateVideoStreamingCapabilities/common.lua [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+local actions = require("user_modules/sequences/actions")
+local runner = require('user_modules/script_runner')
+local hmi_values = require('user_modules/hmi_values')
+local SDL = require('SDL')
+local json = require("modules/json")
+local utils = require("user_modules/utils")
+
+--[[ General configuration parameters ]]
+runner.testSettings.isSelfIncluded = false
+config.defaultProtocolVersion = 2
```
Comment (GetmanetsIrina) [inline test_scripts/UpdateVideoStreamingCapabilities/common.lua:15]:
```suggestion
config.defaultProtocolVersion = 5
```

File: test_scripts/UpdateVideoStreamingCapabilities/001_5_GetCapabilities_invalid_AddVSC.lua [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+checks.invalid_deep_nested_value = common.getVideoStreamingCapability(3)
+checks.invalid_deep_nested_value.additionalVideoStreamingCapabilities[2] = common.getVideoStreamingCapability(1)
+checks.invalid_deep_nested_value.additionalVideoStreamingCapabilities[3] = common.getVideoStreamingCapability(5)
+checks.invalid_deep_nested_value.additionalVideoStreamingCapabilities[3].additionalVideoStreamingCapabilities[2] = common.getVideoStreamingCapability(1)
+checks.invalid_deep_nested_value.additionalVideoStreamingCapabilities[3].additionalVideoStreamingCapabilities[2].additionalVideoStreamingCapabilities[1] = common.getVideoStreamingCapability(3)
+checks.invalid_deep_nested_value.additionalVideoStreamingCapabilities[3].additionalVideoStreamingCapabilities[2].additionalVideoStreamingCapabilities[1].additionalVideoStreamingCapabilities[1] = common.getVideoStreamingCapability(2)
+checks.invalid_deep_nested_value.additionalVideoStreamingCapabilities[3].additionalVideoStreamingCapabilities[2].additionalVideoStreamingCapabilities[1].additionalVideoStreamingCapabilities[1].additionalVideoStreamingCapabilities[2].pixelPerInch = -2 -- invalid value
```
Comment (GetmanetsIrina) [inline test_scripts/UpdateVideoStreamingCapabilities/001_5_GetCapabilities_invalid_AddVSC.lua:37]: Do we really need so many checks for nested structs?
In my opinion, it's enough to check one level in a nested struct to make sure that existed logic works as expected.

File: test_scripts/UpdateVideoStreamingCapabilities/001_1_GetCapabilities_with_addVSC.lua [hunk 4]
```diff
@@ -0,0 +1,34 @@
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0296-Update-video-streaming-capabilities-during-ignition-cycle.md
+--
+-- Description: SDL applies the videoStreamingCapability with additionalVideoStreamingCapabilities parameter
+--  received from HMI in UI.GetCapabilities response
+--
+-- Preconditions:
+-- 1. SDL and HMI are started
+
```
Comment (VjKlepikov) [inline test_scripts/UpdateVideoStreamingCapabilities/001_1_GetCapabilities_with_addVSC.lua:8]:
```suggestion
--
```
I would propose to comment out this line. This comment relates to next scripts too.

File: test_scripts/UpdateVideoStreamingCapabilities/001_7_GetCapabilities_with_addVSC_only.lua [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+-- - a. apply the videoStreamingCapability with only additionalVideoStreamingCapabilities parameter internally
+-- 3. App registers with 5 transport protocol
+-- 4. App requests GetSystemCapability(VIDEO_STREAMING)
+-- SDL does:
+-- - a. send GetSystemCapability response with videoStreamingCapability that contains
+--    only additionalVideoStreamingCapabilities received from HMI in UI.GetCapabilities response
+---------------------------------------------------------------------------------------------------
+-- [[ Required Shared libraries ]]
+local common = require('test_scripts/UpdateVideoStreamingCapabilities/common')
+
```
Comment (VjKlepikov) [inline test_scripts/UpdateVideoStreamingCapabilities/001_7_GetCapabilities_with_addVSC_only.lua:23]:
```suggestion
--[[ Local Variables ]]
```
I would propose to add a comment. This comment relates to next scripts too.

File: test_scripts/UpdateVideoStreamingCapabilities/001_7_GetCapabilities_with_addVSC_only.lua [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+local vsc = {
+    additionalVideoStreamingCapabilities = {}
```
Comment (VjKlepikov) [inline test_scripts/UpdateVideoStreamingCapabilities/001_7_GetCapabilities_with_addVSC_only.lua:25]:
```suggestion
   additionalVideoStreamingCapabilities = {}
```
Please remove redundant a whitespace

File: test_scripts/UpdateVideoStreamingCapabilities/001_1_GetCapabilities_with_addVSC.lua [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+---------------------------------------------------------------------------------------------------
+-- [[ Required Shared libraries ]]
+local common = require('test_scripts/UpdateVideoStreamingCapabilities/common')
+
+--[[ Scenario ]]
+common.Title("Preconditions")
+common.Step("Clean environment", common.preconditions)
+common.Step("Set HMI Capabilities", common.setHMICapabilities)
+common.Step("Start SDL, HMI, connect Mobile, start Session", common.start, { common.hmiDefaultCapabilities })
```
[the remark this answers is not in the document]
Comment (aderiabin) [reply test_scripts/UpdateVideoStreamingCapabilities/001_1_GetCapabilities_with_addVSC.lua:27]: @GetmanetsIrina `common.hmiDefaultCapabilities` contains `additionalVideoStreamingCapabilities` in this case (it is added on previous step)

File: test_scripts/UpdateVideoStreamingCapabilities/common.lua (hunk 2 above)
Comment (aderiabin) [reply test_scripts/UpdateVideoStreamingCapabilities/common.lua]: Updated (see ff7809e)

File: test_scripts/UpdateVideoStreamingCapabilities/001_3_GetCapabilities_in_range_minsize_maxsize_array.lua (hunk 1 above)
Comment (aderiabin) [reply test_scripts/UpdateVideoStreamingCapabilities/001_3_GetCapabilities_in_range_minsize_maxsize_array.lua]: Done (see ff7809e)

File: test_scripts/UpdateVideoStreamingCapabilities/001_5_GetCapabilities_invalid_AddVSC.lua (hunk 3 above)
Comment (aderiabin) [reply test_scripts/UpdateVideoStreamingCapabilities/001_5_GetCapabilities_invalid_AddVSC.lua]: Done (see d0f84bf)

File: test_scripts/UpdateVideoStreamingCapabilities/001_1_GetCapabilities_with_addVSC.lua [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+common.Step("RAI", common.registerAppWOPTU)
+common.Step("Activate App", common.activateApp)
```
[the remark this answers is not in the document]
Comment (aderiabin) [reply test_scripts/UpdateVideoStreamingCapabilities/001_1_GetCapabilities_with_addVSC.lua]: Done (see d0f84bf)

File: test_scripts/UpdateVideoStreamingCapabilities/001_7_GetCapabilities_with_addVSC_only.lua (hunk 5 above)
Comment (aderiabin) [reply test_scripts/UpdateVideoStreamingCapabilities/001_7_GetCapabilities_with_addVSC_only.lua:23]: Done (see ff7809e)

File: test_scripts/UpdateVideoStreamingCapabilities/001_7_GetCapabilities_with_addVSC_only.lua (hunk 6 above)
Comment (aderiabin) [reply test_scripts/UpdateVideoStreamingCapabilities/001_7_GetCapabilities_with_addVSC_only.lua]: Done (see 38967ac)

File: test_scripts/UpdateVideoStreamingCapabilities/001_1_GetCapabilities_with_addVSC.lua (hunk 4 above)
Comment (aderiabin) [reply test_scripts/UpdateVideoStreamingCapabilities/001_1_GetCapabilities_with_addVSC.lua]: Done (see 38967ac)

**Pull request merged**